app/uk/gov/hmrc/eoricommoncomponent/frontend/services/UserGroupIdSubscriptionStatusCheckService.scala

```diff
@@ -30,27 +29,28 @@ import scala.concurrent.{ExecutionContext, Future}
 @Singleton
 class UserGroupIdSubscriptionStatusCheckService @Inject() (
   subscriptionStatusService: SubscriptionStatusService,
-  save4LaterConnector: Save4LaterConnector
+  save4Later: Save4LaterService
 )(implicit ec: ExecutionContext) {
   private val idType = "SAFE"
 
   def checksToProceed(groupId: GroupId, internalId: InternalId, service: Service)(continue: => Future[Result])(
     userIsInProcess: => Future[Result]
   )(otherUserWithinGroupIsInProcess: => Future[Result])(implicit hc: HeaderCarrier): Future[Result] =
```

Comment:
Logic.
When there is a cacheId for the group then there was a previous application.
If this is a new application for a different service than last time clear the email from the users cache